tests/integrations/mcs/resourcemanager/service_limit_test.go

```diff
@@ -0,0 +1,167 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcemanager_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/pingcap/failpoint"
+	rmpb "github.com/pingcap/kvproto/pkg/resource_manager"
+
+	pd "github.com/tikv/pd/client"
+	"github.com/tikv/pd/client/pkg/caller"
+	"github.com/tikv/pd/client/resource_group/controller"
+	"github.com/tikv/pd/pkg/keyspace"
+	"github.com/tikv/pd/pkg/mcs/resourcemanager/server"
+	"github.com/tikv/pd/tests"
+)
+
+const testKeyspaceName = "test_keyspace"
+
+type serviceLimitTestSuite struct {
+	suite.Suite
+	ctx        context.Context
+	cancel     func()
+	cluster    *tests.TestCluster
+	leader     *tests.TestServer
+	client     pd.Client
+	controller *controller.ResourceGroupsController
+	keyspaceID uint32
+}
+
+func TestServiceLimitTestSuite(t *testing.T) {
+	suite.Run(t, new(serviceLimitTestSuite))
+}
+
+func (suite *serviceLimitTestSuite) SetupTest() {
+	re := suite.Require()
+	re.NoError(failpoint.Enable("github.com/tikv/pd/pkg/keyspace/skipSplitRegion", "return(true)"))
+	defer func() {
+		re.NoError(failpoint.Disable("github.com/tikv/pd/pkg/keyspace/skipSplitRegion"))
+	}()
+
+	suite.ctx, suite.cancel = context.WithCancel(context.Background())
+	cluster, err := tests.NewTestCluster(suite.ctx, 1)
+	re.NoError(err)
+	suite.cluster = cluster
+	re.NoError(cluster.RunInitialServers())
+	suite.leader = suite.cluster.GetServer(suite.cluster.WaitLeader())
+	re.NotNil(suite.leader)
+	// Create a keyspace.
+	meta, err := suite.leader.GetKeyspaceManager().CreateKeyspace(
+		&keyspace.CreateKeyspaceRequest{
+			Name: testKeyspaceName,
+		},
+	)
+	re.NoError(err)
+	suite.keyspaceID = meta.GetId()
+	// Prepare the client and controller.
+	suite.client, err = pd.NewClientWithContext(
+		suite.ctx,
+		caller.TestComponent,
+		suite.cluster.GetConfig().GetClientURLs(),
+		pd.SecurityOption{})
+	re.NoError(err)
+	waitLeaderServingClient(re, suite.client, suite.leader.GetAddr())
+	suite.controller, err = controller.NewResourceGroupController(suite.ctx, 1, suite.client, nil, suite.keyspaceID)
+	re.NoError(err)
+	suite.controller.Start(suite.ctx)
+}
+
+func (suite *serviceLimitTestSuite) TearDownTest() {
+	suite.cancel()
+	suite.cluster.Destroy()
+}
+
+func (suite *serviceLimitTestSuite) TestKeyspaceServiceLimit() {
+	re := suite.Require()
+
+	ctx, cancel := context.WithCancel(suite.ctx)
+	defer cancel()
+	const requestRU = 200.
+	tokenBucketRequest := &rmpb.TokenBucketsRequest{
+		Requests: []*rmpb.TokenBucketRequest{
+			{
+				ResourceGroupName: server.DefaultResourceGroupName,
```

Comment:
Do we need to test other non-default groups?

Reply:
I will add more test cases later. This PR primarily focuses on establishing a test framework and implementing some minor fixes.